internal/sub_reconcilers/reconcile_deployment_updates.go

```diff
@@ -115,21 +115,25 @@ func (r *DeploymentImageReconciler) upgradeOperand(ctx context.Context, bestie *
 	}
 
 	// compare current container image to spec image
-	bestieImageDifferent := !reflect.DeepEqual(dp.Spec.Template.Spec.Containers[0].Image, getBestieContainerImage(bestie))
-
-	if bestieImageDifferent {
-		if bestieImageDifferent {
-			r.Log.Info("Upgrade Operand")
-			dp.Spec.Template.Spec.Containers[0].Image = getBestieContainerImage(bestie)
-		}
-		err = r.client.Update(ctx, dp)
-		if err != nil {
-			r.Log.Error(err, "Deployment failed.")
-			return err
+	imageInDeployment := "unknown"
+	positionInContainerArray := 0
+	for pos, container := range dp.Spec.Template.Spec.Containers {
+		if container.Name == "bestie" {
+			imageInDeployment = container.Image
+			positionInContainerArray = pos
 		}
-		// Level 4 Add metrics
-		bestie_metrics.ApplicationUpgradeCounter.Inc()
 	}
+	if imageInDeployment != getBestieContainerImage(bestie) {
+		r.Log.Info("Upgrade Operand")
```

Comment:
Could you please specify which operand is getting upgraded?

Reply:
mmm there is only one.. but I guess can use a different word than operand